internal/topology/check/compatibility_test.go

```diff
@@ -120,13 +120,12 @@ func TestObjectsAreCompatible(t *testing.T) {
 	for _, tt := range referencedObjectsCompatibilityTestCases {
 		t.Run(tt.name, func(t *testing.T) {
 			g := NewWithT(t)
-
 			allErrs := ObjectsAreCompatible(tt.current, tt.desired).ToAggregate()
 			if tt.wantErr {
-				g.Expect(allErrs).To(HaveOccurred())
+				g.Expect(allErrs).ToNot(BeEmpty())
 				return
 			}
-			g.Expect(allErrs).ToNot(HaveOccurred())
+			g.Expect(allErrs).To(BeNil())
```

Comment:
Would this also work with `g.Expect(allErrs).To(BeEmpty())` It just feels fragile to me to rely on the array being nil vs it being empty (aka nil or an array with len 0)

Reply:
Unfortunately, BeEmpty does not work in case allErrs is nil

```
BeEmpty matcher expects a string/array/map/channel/slice.  Got:
            <nil>: nil
```

Reply:
Fun, that almost sounds like a gomega bug as len() also works on nil, but of course find to keep as is :)

Reply:
I think the issue is the `ToAggregate()` which converts the list into a single error or nil.

I think we should either:
* use `ToAggregate()` and check against nil / not nil via our regular `HaveOccurred` matcher (using BeEmpty() for a nil check seems not ideal to me)
    * ideally we would rename the var from allErrs to just err when using `ToAggregate()` to avoid misunderstandings
* don't use `ToAggregate()` and use `To(BeEmpty())` and `Not(BeEmpty())`

I think the misleading variable name is what lead to this change :)

Reply:
fixed